### DISCUSSION TITLE
how can we have an input with multiline?

### DISCUSSION THREAD

**Discussion by pheninux** [charmbracelet/bubbletea#306] [Q&A] [answered]
Hello ,
how can we have an input with multiline or a way of writing on several lines ?
thanks in advance

**Comment by Robert-M-Muench** [comment]
I'm interested in this too and am currently researching how to do it. My current best bet is to use a simple terminal editor and re-work it into a bubbletea component. A good candidate could be: https://github.com/hibiken/mini/blob/master/mini.go

The idea is, to switch into editor mode, and return to the application GUI after editing. Maybe it's possible to render the editor to a fixed area while still seeing the rest of the application.

**Reply by pheninux** [reply]
hello , sorry for my late response  , i appreciate your help but i cant imagine how can i do this ! have you some exemple plz

**Reply by Robert-M-Muench** [reply]
It's just an idea that requires some work. Nothing is done yet. Hence, I don't have an example.

**Comment by CalebJohnHunt** [comment, marked as answer]
I know it's been quite a while, and I hope you've found your answer. For anyone still looking, the companion package [Bubbles](https://github.com/charmbracelet/bubbles) has many components which work well with Bubbletea. In this case, take a look at their [text area](https://github.com/charmbracelet/bubbles#text-area) component.
